packages/hardhat-core/src/internal/core/runtime-environment.ts

```diff
@@ -267,16 +275,28 @@ export class Environment implements HardhatRuntimeEnvironment {
     globalAsAny.runSuper = runSuper;
 
     let modifiedHreWithParentTaskProfile: any | undefined;
-    if (this.hardhatArguments.flamegraph === true) {
+    if (
+      this.hardhatArguments.flamegraph === true ||
+      subtaskArguments !== undefined
+    ) {
```

Comment:
Hmm, I don't think this is right. This seems to mean that we will generate a flamegraph when subtask arguments are defined, which we don't want to do.

Reply:
I guess we can always make the `subtaskArguments !== undefined`, by defaulting to `{}`